src/psyclone/psyad/transformations/preprocess.py

```diff
@@ -56,6 +56,15 @@ def preprocess_trans(kernel_psyir):
     '''
     dot_product_trans = DotProduct2CodeTrans()
     matmul_trans = Matmul2CodeTrans()
+    arrayrange2loop_trans = ArrayRange2LoopTrans()
+
+    # Replace array-ranges with explicit loops
+    for assignment in kernel_psyir.walk(Assignment):
+        while True:
```

Comment:
Please add a comment about this `while True` as it looks very odd - presumably it's here because the transformation does one array index at a time?

Reply:
Added a comment. Yes you are right. There is a nemo-specific version that operates on the whole assignment but just the single generic transformation.